src/Lime.Protocol/LimeUri.cs

```diff
@@ -33,9 +34,24 @@ public LimeUri(string uriPath)
             }
             else if (!Uri.IsWellFormedUriString(uriPath, UriKind.Relative))
             {
-                throw new ArgumentException("Invalid URI format");
+                // TODO: This 'if' statement is only necessary while the related issue is not fixed
+                // Issue: https://github.com/dotnet/runtime/issues/21626
+                if (Uri.TryCreate(uriPath, UriKind.Absolute, out var receivedUri) &&
+                    ReceivedUriPathIsEncoded(receivedUri) &&
```

Comment:
We could use uriPath directly here - since receivedUri.OriginalString is used inside it.